code/obj/item/implant.dm

```diff
@@ -890,6 +896,366 @@ ABSTRACT_TYPE(/obj/item/implant/revenge)
+				return
+
+		src.heat_dissipation = initial(src.heat_dissipation)
+
+		var/fail_reason
+		if (!ismob(src.owner))
+			fail_reason = MARIONETTE_IMPLANT_ERROR_NO_TARGET
+			send_activation_reply(signal.data["sender"], fail_reason)
+			return
+
+		var/mob/living/carbon/human/H = src.owner
+		if (istype(H) && H.decomp_stage != DECOMP_STAGE_NO_ROT)
+			fail_reason = MARIONETTE_IMPLANT_ERROR_DEAD_TARGET
+			send_activation_reply(signal.data["sender"], fail_reason)
+			return
+
+		var/data = signal.data["data"]
+		switch (command)
+			if ("say", "speak")
+				if (!isdead(src.owner))
+					logTheThing(LOG_COMBAT, src.owner, "was forced by \a [src] to say \"[data]\" at [log_loc(src.owner)] (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+					data = copytext(strip_prefix(data, "*"), 1, 46) // Trim starting asterisks to prevent force-emoting
+					src.owner.say(data)
+				else
+					fail_reason = MARIONETTE_IMPLANT_ERROR_DEAD_TARGET
+				src.adjust_heat(15)
+			if ("emote")
+				data = lowertext(data)
+				if (!isdead(src.owner))
+					if (data in src.emote_blacklist)
+						fail_reason = MARIONETTE_IMPLANT_ERROR_INVALID
+					else
+						logTheThing(LOG_COMBAT, src.owner, "was forced by \a [src] to emote \"[data]\" at [log_loc(src.owner)] by (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+						src.owner.emote(data)
+				else
+					fail_reason = MARIONETTE_IMPLANT_ERROR_DEAD_TARGET
+				src.adjust_heat(15)
+			if ("move", "step", "bump")
+				logTheThing(LOG_COMBAT, src.owner, "was forced by \a [src] to step to the [lowertext(data)] at [log_loc(src.owner)] (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+				var/step_dir = text2dir(uppertext(data))
+				if (step_dir && (step_dir in cardinal))
+					step(src.owner, step_dir)
+				else
+					fail_reason = MARIONETTE_IMPLANT_ERROR_INVALID
+				src.adjust_heat(5)
+			if ("shock", "zap")
+				logTheThing(LOG_COMBAT, src.owner, "was shocked by \a [src] at [log_loc(src.owner)] (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+				boutput(src.owner, SPAN_ALERT("You feel a shock from inside your body!"))
+				src.owner.do_disorient(90, knockdown = 7 SECONDS, disorient = 3 SECONDS)
+				playsound(src.owner, 'sound/impact_sounds/Energy_Hit_3.ogg', 20, TRUE, -1)
+				src.adjust_heat(50)
+			if ("drop", "release")
+				if (!isdead(src.owner))
+					var/obj/item/I = src.owner.equipped()
+					if (istype(I))
+						logTheThing(LOG_COMBAT, src.owner, "was forced to drop \the [I] by \a [src] at [log_loc(src.owner)] (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+						boutput(src.owner, SPAN_ALERT("Your grip on \the [I] suddenly relaxes!"))
+						H.drop_item()
+					else
+						fail_reason = MARIONETTE_IMPLANT_ERROR_INVALID
+					src.adjust_heat(60)
```

Comment:
Since these values are so high that just using some of the functions 2 or 3 times could burn out the implant it would be nice if they showed up on the remote as well as the wall of text manual. Maybe just a "low/medium/high" heat warning when you have an action selected?